hal/ti_hercules.c

```diff
@@ -0,0 +1,159 @@
+/* ti_hercules.c
+ *
+ * Copyright (C) 2021 wolfSSL Inc.
+ *
+ * This file is part of wolfBoot.
+ *
+ * wolfBoot is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfBoot is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#include <stdint.h>
+#include <string.h>
+#include <target.h>
+#include "image.h"
+#include "hal.h"
+
+#ifndef CORTEX_R5
+#   error "wolfBoot TI Hercules HAL: wrong architecture selected. Please compile with TARGET=ti_hercules."
+#endif
+
+#include <F021.h>
+#include <stdio.h>
+
+//#define FLASH_DEMO 1
+
+/* public HAL functions */
+void hal_init(void)
+{
+    int freq_MHz = 16;
+    int st = Fapi_initializeFlashBanks(freq_MHz);
+    if (st != 0) {
+        printf("Failed Fapi_initializeFlashBanks(%d) => (%d)\n", freq_MHz, st);
+        return;
+    }
+
+#if FLASH_DEMO
```

Comment:
This could not be defined. Prefer `#if defined(FLASH_DEMO) && FLASH_DEMO`

Reply:
Done